R/arc-open.R

```diff
@@ -112,3 +149,59 @@ arc_open <- function(url, token = arc_token()) {
 
   res
 }
+
+#' Test if the vector is a content url of the supplied type
+#' @keywords internal
+is_arc_content_url <- function(x, type = c("search", "item", "group", "user")) {
+  pattern <- paste0("arcgis\\.com/home/(", paste0(type, collapse = "|"), ")\\.html")
+  is.character(x) & grepl(pattern, x, perl = TRUE)
+}
+
+#' Convert a URL into a content URL
+#' @keywords internal
+arc_url_item <- function(
+    id,
+    key = "id",
+    base_url = "https://www.arcgis.com/sharing/rest/content",
```

Comment:
Please use [`arc_host()`](https://developers.arcgis.com/r-bridge/api-reference/arcgisutils/arc_host.html) instead of hard coding this.